File: 0841-keys-and-rooms/0841-keys-and-rooms.js
/**
 * @param {number[][]} rooms
 * @return {boolean}
 */
const canVisitAllRooms = function(rooms) {
    const visited = new Set();
  
    dfs(rooms, 0, visited);

    return visited.size === rooms.length;
};

const dfs = (graph, node, visited) => {
    visited.add(node);

    for (let i = 0; i < graph[node].length; i++) {
        if (visited.has(graph[node][i])) continue;
        
        dfs(graph, graph[node][i], visited);
    }
}

File: 0841-keys-and-rooms/0841-keys-and-rooms.py
from collections import deque

class Solution(object):
    def canVisitAllRooms(self, rooms):
        """
        :type rooms: List[List[int]]
        :rtype: bool
        """
        N = len(rooms)
        visited = [False] * N
        q = deque([0])

        while q:
            node = q.popleft()
            visited[node] = True

            for neighbor in rooms[node]:
                if not visited[neighbor]:
                    q.append(neighbor)  

        return len([i for i in visited if i]) == N